src/hera/workflows/script.py

```diff
@@ -669,7 +678,7 @@ def generate_source(self, instance: Script) -> str:
         # in order to have consistent looking functions and getting rid of any comments
         # parsing issues.
         # See https://github.com/argoproj-labs/hera/issues/572
-        content = roundtrip(inspect.getsource(instance.source)).splitlines()
+        content = roundtrip(textwrap.dedent(inspect.getsource(instance.source))).splitlines()
```

Comment:
I think this also fixes being able to define normal script functions within tests themselves! 👀